src/Cardano/Metadata/Validation/Wallet.hs

```diff
@@ -0,0 +1,91 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+{- | Wallet-specific validation rules.
+-}
+
+module Cardano.Metadata.Validation.Wallet
```

Comment:
I don't get this module's name? There's nothing specific about _the wallet_ per se. If this refers to well-known properties, then we should refer to them as "well known properties" :no_mouth: That the wallet relies on well-known properties is anecdotal, many tools will eventually rely on well-known properties too.

Reply:
You have noticed that I have not re-written the existing wallet
validation logic in the style of the validation library provided by
the "metadata-lib" project. If I was to do so, we may well find parts
of the logic that can be shared with other projects, just like some of
the rules I'm already sharing from the "metadata-lib" project (see
the contents of "defaultRules"). I haven't re-written it for a number of reasons
though:
  1. It works
  2. I have limited time
  3. There aren't enough tests around it for me to be confident my
     changes won't introduce an error.

I think I've already done a great deal to introduce a framework around
which we can share rules in the "metadata-lib" project. Further work
can certainly generalise and share some of the wallet's validation
logic, now there is a framework for doing so.

So yeah, you're correct, this logic is not necessarily
wallet-specific. There is some logic in your work that can be shared
with other projects. But from my point of view "validateEntry" is a
black-box which is used to validate wallet entries and cannot be used
to validate non-wallet entries (for example, a metadata entry without
a policy would not pass "validateEntry", and policies are very
specific to the minting of assets).